common/src/stack/command/stack/commands/load/json/plugin_software.py

```diff
@@ -0,0 +1,158 @@
+# @copyright@
+# Copyright (c) 2006 - 2018 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import stack.commands
+import json
+from stack.exception import CommandError
+import subprocess
+
+class Plugin(stack.commands.Plugin, stack.commands.Command):
+	notifications = True
+
+	def provides(self):
+		return 'software'
+
+
+	def run(self, args):
+
+		# check if the user would like to import software data
+		# if there are no args, assume the user would like to import everthing
+		if args and 'software' not in args:
+			return
+
+		# self.owner.data contains the data from the json file defined in init
+		# check if there is any software data before we go getting all kinds of key errors
+		if 'software' in self.owner.data:
+			import_data = self.owner.data['software']
+		else:
+			self.owner.log.info('no software data in json file')
+			return
+
+		self.notify('\n\tLoading software\n')
+		# check to make sure 'box' 'pallet' and 'cart' all exist first to avoid a key error
+		# add boxes
+		if import_data['box']:
+			for box in import_data['box']:
+				parameters = [
+					f'{box["name"]}',
+					f'os={box["os"]}',
+				]
+				self.owner.try_command('add.box', parameters, f'adding box {box["name"]}', 'exists')
+
+		# add pallets
+		if import_data['pallet']:
+			# create a list of pallets to add, excluding ones already present on the system
+			pallets_to_add = []
+			existing_pallet_data = json.loads(self.owner.command('list.pallet', [ 'output-format=json', 'expanded=true' ]))
+
+
+			# iterate through each pallet, and only add it if not aleady on the system
+			for pallet in import_data['pallet']:
+				for existing_pallet in existing_pallet_data:
+					if (pallet['name'] == existing_pallet['name'] and
+					    pallet['version'] == existing_pallet['version'] and
+					    pallet['release'] == existing_pallet['release'] and
+					    ' '.join(pallet['boxes']) == existing_pallet['boxes']):
```

Comment:
The list of boxes (and thus, string) is order dependent, no?  Perhaps converting this to a set and testing sets for equality would be better?